tests/CacheTest.php

```diff
@@ -16,83 +14,84 @@ protected function setUp(): void
         $this->cache = new Cache();
     }
 
+    /**
+     * @throws \InvalidArgumentException|InvalidArgumentException
+     */
     public function testGetInvalidCacheKey()
     {
-        $this->expectException(CacheException::class);
+        $this->expectException(\InvalidArgumentException::class);
         $this->cache->get('');
     }
 
+    /**
+     * @throws \InvalidArgumentException|InvalidArgumentException
+     */
     public function testSetInvalidCacheKey()
     {
-        $this->expectException(CacheException::class);
+        $this->expectException(\InvalidArgumentException::class);
         $this->cache->set('', 'a', 100);
     }
 
     /**
-     * @throws CacheException
+     * @throws InvalidArgumentException
      */
     public function testGetNonExistent()
     {
         $this->assertNull($this->cache->get('random'));
     }
 
     /**
-     * @throws CacheException
+     * @throws InvalidArgumentException
      */
     public function testSetBoolean()
     {
         $this->cache->set('isMobile', true, 100);
-        $this->assertInstanceOf(CacheItem::class, $this->cache->get('isMobile'));
-        $this->assertTrue($this->cache->get('isMobile')->get());
+        $this->assertTrue($this->cache->get('isMobile'));
 
         $this->cache->set('isTablet', false, 100);
-        $this->assertInstanceOf(CacheItem::class, $this->cache->get('isTablet'));
-        $this->assertFalse($this->cache->get('isTablet')->get());
+        $this->assertFalse($this->cache->get('isTablet'));
     }
 
     /**
-     * @throws CacheException
+     * @throws InvalidArgumentException
      */
     public function testGetTTL0()
     {
         $this->cache->set('isMobile', true, 0);
-        $this->assertInstanceOf(CacheItem::class, $this->cache->get('isMobile'));
-        $this->assertNull($this->cache->get('isMobile')->expiresAt);
-        $this->assertNull($this->cache->get('isMobile')->expiresAfter);
+        $this->assertNull($this->cache->get('isMobile'));
     }
 
-    public function testGetTtlIsInteger()
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function testSetTtlAsInteger()
```

Comment:
[Nice-to-have]
You use `TTL` in some methods while in some `Ttl`. It'd be nice to have some consistency.